Src/libCZI/CziParse.cpp

```
@@ -201,45 +201,49 @@ using namespace libCZI;
 
-    for (int i = 0; i < attachmentDirSegment.data.EntryCount; ++i)
-    {
-        ConvertToHostByteOrder::Convert(pBuffer.get() + i);
+        // now read the attachment-entries
+        try
+        {
+            str->Read(offset + sizeof(attachmentDirSegment), pBuffer.get(), attachmentEntriesSize, &bytesRead);
+        }
+        catch (const std::exception&)
+        {
+            std::throw_with_nested(LibCZIIOException("Error reading FileHeaderSegment", offset + sizeof(attachmentDirSegment), attachmentEntriesSize));
+        }
 
-        const AttachmentEntryA1* pSrc = pBuffer.get() + i;
-        CCziAttachmentsDirectoryBase::AttachmentEntry ae;
-        bool b = CheckAttachmentSchemaType((const char*)&pSrc->SchemaType[0], 2);
-        if (b == false)
+        if (bytesRead != attachmentEntriesSize)
         {
-            // TODO: what do we do if we encounter this...?
-            continue;
+            CCZIParse::ThrowNotEnoughDataRead(offset + sizeof(attachmentDirSegment), attachmentEntriesSize, bytesRead);
         }
 
-        ae.FilePosition = pSrc->FilePosition;
-        ae.ContentGuid = pSrc->ContentGuid;
-        memcpy(&ae.ContentFileType[0], &pSrc->ContentFileType[0], sizeof(AttachmentEntryA1::ContentFileType));
-        memcpy(&ae.Name[0], &pSrc->Name[0], sizeof(AttachmentEntryA1::Name));
-        ae.Name[sizeof(AttachmentEntryA1::Name) - 1] = '\0';
-        addFunc(ae);
+        for (int i = 0; i < attachmentDirSegment.data.EntryCount; ++i)
+        {
+            ConvertToHostByteOrder::Convert(pBuffer.get() + i);
+
+            const AttachmentEntryA1* pSrc = pBuffer.get() + i;
+            CCziAttachmentsDirectoryBase::AttachmentEntry ae;
+            bool b = CheckAttachmentSchemaType(reinterpret_cast<const char*>(&pSrc->SchemaType[0]), 2);
+            if (b == false)
+            {
+                // TODO: what do we do if we encounter this...?
```

Is the todo still relevant (as in do we need an explicit communication via exception to demark this situation) or is it sufficient to continue?

---

afair - it is still relevant, and - ja, to be decided if this is considered an error or not. For the time being it is silently ignored, which is probably the worst thing one can do - but which causes least work.
So, yes, still relevant I guess, but I am not aware of any past problems here, so - as low a priority as can,